indico_zodbimport/modules/event_papers.py

```diff
@@ -169,6 +223,87 @@ def _migrate_paper_roles(self, contribution):
+                # ... except if said revision has been judged before being marked as submitted (!)
+                if old_judgment._submitted:
+                    self.importer.print_warning(cformat('%{yellow!}Revision judged without being submitted! [{}: {}]')
+                                                .format(contribution, old_revision._version), event_id=self.event.id)
+                else:
+                    continue
+
+            # It seems contradictory, but 'submitted' in legacy means that there is a final decision
+            # We'll ignore legacy custom states and use TBC
+            state = (JUDGMENT_STATE_REVISION_MAP.get(int(old_judgment._judgement._id),
+                                                     PaperRevisionState.to_be_corrected)
+                     if old_judgment._submitted
+                     else PaperRevisionState.submitted)
+            judge = old_judgment._author.user if old_judgment._submitted else None
+            judgment_dt = _to_utc(old_judgment._submissionDate) if old_judgment._submitted else None
+            last_file = (LegacyPaperFile.query.with_parent(contribution)
+                         .filter(LegacyPaperFile.revision_id == old_revision._version)
+                         .order_by(LegacyPaperFile.created_dt.desc())
+                         .limit(1)).first()
+            # Legacy didn't keep track of the submission date (nor submitter for that matter)
+            # we're taking the most recent uploaded file and using that one
+            # if there are no files, the event's end date will be used
+            submitted_dt = _to_utc(last_file.created_dt) if last_file else self.event.end_dt
+            revision = PaperRevision(state=state, submitter=self.janitor_user, judge=judge,
+                                     judgment_dt=judgment_dt, judgment_comment=old_judgment._comments,
+                                     submitted_dt=submitted_dt)
+            self.legacy_contrib_revision_map[(contribution.id, old_revision._version)] = revision
+
+            # Then we'll add the layout and content reviews
+            review_colors = ''
+
+            for old_content_review in old_content_reviews:
+                if old_content_review._submitted:
+                    review = self._migrate_review(contribution, old_content_review, PaperReviewType.content)
+                    revision.reviews.append(review)
+                    review_colors += _review_color(review, 'C')
+            if old_layout_review._submitted:
+                review = self._migrate_review(contribution, old_layout_review, PaperReviewType.layout)
+                revision.reviews.append(review)
+                review_colors += _review_color(review, 'L')
+            contribution.paper_revisions.append(revision)
+
+            self.importer.print_info(cformat('\tRevision %{{blue!}}{}%{{reset}} %{{white,{}}}  %{{reset}} {}'.format(
+                n, STATE_COLOR_MAP[state], review_colors)))
```

Comment:
I'd `cformat(...).format(n)` (even though it doesn't make a real difference here since it's just a number which won't ever interfere with the cformat logic)